frontends/p4/def_use.h

```diff
@@ -555,63 +647,8 @@ class ComputeWriteSet : public Inspector, public IHasDbPrint {
-          currentDefinitions(nullptr),
-          returnedDefinitions(nullptr),
-          exitDefinitions(new Definitions()),
-          storageMap(allDefinitions->storageMap),
-          lhs(false),
-          virtualMethod(false) {
-        CHECK_NULL(allDefinitions);
-        visitDagOnce = false;
-    }
-
-    // expressions
-    bool preorder(const IR::Literal *expression) override;
-    bool preorder(const IR::Slice *expression) override;
-    bool preorder(const IR::TypeNameExpression *expression) override;
-    bool preorder(const IR::PathExpression *expression) override;
-    bool preorder(const IR::Member *expression) override;
-    bool preorder(const IR::ArrayIndex *expression) override;
-    bool preorder(const IR::Operation_Binary *expression) override;
-    bool preorder(const IR::Mux *expression) override;
-    bool preorder(const IR::SelectExpression *expression) override;
-    bool preorder(const IR::ListExpression *expression) override;
-    bool preorder(const IR::Operation_Unary *expression) override;
-    bool preorder(const IR::MethodCallExpression *expression) override;
-    bool preorder(const IR::DefaultExpression *expression) override;
-    bool preorder(const IR::Expression *expression) override;
-    bool preorder(const IR::InvalidHeader *expression) override;
-    bool preorder(const IR::InvalidHeaderUnion *expression) override;
-    bool preorder(const IR::P4ListExpression *expression) override;
-    bool preorder(const IR::HeaderStackExpression *expression) override;
-    bool preorder(const IR::StructExpression *expression) override;
-    // statements
-    bool preorder(const IR::P4Parser *parser) override;
-    bool preorder(const IR::P4Control *control) override;
-    bool preorder(const IR::P4Action *action) override;
-    bool preorder(const IR::P4Table *table) override;
-    bool preorder(const IR::Function *function) override;
-    bool preorder(const IR::AssignmentStatement *statement) override;
-    bool preorder(const IR::ReturnStatement *statement) override;
-    bool preorder(const IR::ExitStatement *statement) override;
-    bool preorder(const IR::BreakStatement *statement) override;
-    bool handleJump(const char *tok, Definitions *&defs);
-    bool preorder(const IR::ContinueStatement *statement) override;
-    bool preorder(const IR::IfStatement *statement) override;
-    bool preorder(const IR::ForStatement *statement) override;
-    bool preorder(const IR::ForInStatement *statement) override;
-    bool preorder(const IR::BlockStatement *statement) override;
-    bool preorder(const IR::SwitchStatement *statement) override;
-    bool preorder(const IR::EmptyStatement *statement) override;
-    bool preorder(const IR::MethodCallStatement *statement) override;
-
-    const LocationSet *writtenLocations(const IR::Expression *expression) {
-        expression->apply(*this);
-        return getWrites(expression);
-    }
+ private:
+    std::set<loc_t> &cached_locs;
```

Comment:
why this is `set`? Do you need stable iteration order?

Reply:
changed to `std::unordered_set`